### PR TITLE
feat(user-auth): add changePassword mutation

### PR DESCRIPTION
**Pull request by khanjasir90**
## #75
Signed-off-by: khanjasir90 <khanmohdjasir@gmail.com>
Description: added changePassword Mutation

### CODE REVIEW

**Comment by khanjasir90** [comment]
@d-e-v-esh could u pls review it

File: src/resolvers/userResolver.js [hunk 1]
```diff
@@ -231,6 +231,58 @@ const userResolver = {
       );
     },
 
+    changePassword: async(_, { oldPassword, newPassword }, { req }) => {
+      const { password } = await User.findById(req.session.userId);
+      const isOldPassword = await bcrypt.compare(oldPassword, password);
```
Comment (d-e-v-esh) [inline src/resolvers/userResolver.js]: The variable `isOldPassword` can be more descriptive so it is easier to read. You might write it as `isOldPasswordCorrect` or something like it.

File: src/resolvers/userResolver.js [hunk 2]
```diff
@@ -231,6 +231,58 @@ const userResolver = {
...
+          errors: [
+            {
+              field: "password",
+              message: "old password does not match",
+            },
+          ],
+        };
+      }
+
+      if(oldPassword === newPassword) {
```
Comment (d-e-v-esh) [inline src/resolvers/userResolver.js]: I don't think we need to give them an error if both passwords are the same so we can remove this check.

**Review by d-e-v-esh — CHANGES_REQUESTED** [review]
The code looks really good 💯. Just a couple of things that we need to change.

**Comment by khanjasir90** [comment]
@d-e-v-esh I have made the required changes!

**Comment by d-e-v-esh** [comment]
The code works well, clean up the extra commit and I'll merge. 👍

**Pull request merged**